### ISSUE TITLE
obj_loss, precision and recall gets stuck on 0.

### ISSUE THREAD

**Issue by salkhokhar** [meituan/YOLOv6#380]
I was training the yolov6s network on a custom dataset. The precision and recall never rise above 0. The obj_loss gets stuck at nan from the 3rd epoch onwards.
I tried to use "loss = torch.nan_to_num(loss)" as suggested for [the iou loss] (https://github.com/meituan/YOLOv6/issues/319) being stuck at nan.
I used the same data for training with yolov5 and it was working fine.

![Capture](https://user-images.githubusercontent.com/61519106/182066000-88325552-7b19-44d7-bdc7-ea943b1a9ad2.PNG)

**Comment by salkhokhar** [comment]
After messing around with loss functions for hours, the problem seemed to persist.
As it turned out changing the optimizer to adam solved the problem. The algorithm may have stuck on some local minima and SGD may not have been able to get out of it.

**Comment by salkhokhar** [comment]
After messing around with loss functions for hours, the problem seemed to persist.
As it turned out changing the optimizer to adam solved the problem. The algorithm may have stuck on some local minima and SGD may not have been able to get out of it.
